### PR TITLE
Fix and improve DGX tests

### PR DESCRIPTION
**Pull request by pentschev**
Pass `UCX_NET_DEVICES` to scheduler and client processes, as RDMACM requires an IB device to work. Ensures that `UCX_NET_DEVICES` test also works for systems with many GPUs, such as a DGX-2. Add support for DGX A100 where already possible.

### CODE REVIEW

File: dask_cuda/tests/test_dgx.py [hunk 1]
```diff
@@ -272,11 +277,16 @@ def _test_dask_cuda_worker_ucx_net_devices(enable_rdmacm):
...
                 start = time()
                 while True:
-                    if len(client.scheduler_info()["workers"]) == get_gpu_count():
+                    n_gpus = get_n_gpus()
+                    if len(client.scheduler_info()["workers"]) == n_gpus:
                         break
-                    else:
-                        assert time() - start < 10
-                        sleep(0.1)
+                    elif time() - start > 2 * n_gpus:
```
Comment (jakirkham) [inline dask_cuda/tests/test_dgx.py:283]: How did we determine this amount of time?

Comment (pentschev) [reply dask_cuda/tests/test_dgx.py:283]: Trial and error, this seems to be enough for a DGX-2, but the previous 10 seconds that was hardcoded wasn't.

**Comment by pentschev** [comment]
All tests should now pass (or get skipped) on DGX-1, DGX-2 and DGX A100. Could you take another look when you have a chance @jakirkham ?

**Comment by jakirkham** [comment]
Thanks Peter! This seems sensible. Though it appears we are seeing CI failures here. Do we know what might be happening?

**Comment by pentschev** [comment]
> Thanks Peter! This seems sensible. Though it appears we are seeing CI failures here. Do we know what might be happening?

Not really, I can't reproduce this locally, I'll revert the change that is causing that and try again.

**Comment by jakirkham** [comment]
Debugging with a dummy PR ( https://github.com/rapidsai/dask-cuda/pull/319 )

**Comment by pentschev** [comment]
rerun tests

**Comment by pentschev** [comment]
Thansk @jakirkham , merging before something else breaks! 😄

**Pull request merged**